py/server/tests/test_udf_scalar_args.py

```diff
@@ -613,6 +613,12 @@ def f(p1: float, p2: np.float64) -> bool:
         self.assertRegex(str(w[-1].message), "numpy scalar type.*is used")
         self.assertEqual(10, t.to_string().count("true"))
 
+    def test_no_signature(self):
+        builtin_max = max
+        t = empty_table(10).update("X = (int) builtin_max(1, 2, 3)")
```

Comment:
This PR is marked as fixing https://github.com/deephaven/deephaven-core/issues/6349.  #6349 mentions two problems.  This test is casting the return type, so it is either a partial fix of the ticket or there should be another test to check type inference.

Reply:
Looking into this problem some more, `pybind11` actually does provide signature information in a way that I think we are not yet supporting.  From the example in #6349, I added:
```python
print(blackscholes.price.__doc__)
```
This outputs a docstring that is prefixed with the method signature.
```
price(arg0: float, arg1: float, arg2: float, arg3: float, arg4: float, arg5: bool, arg6: bool) -> float
```

Here are more things that may be useful:
```python
print(type(blackscholes.price))
```
```
<class 'builtin_function_or_method'>
```
```python
print(blackscholes.price.__qualname__)
```
```
PyCapsule.price
```

Reply:
1. I don't think the doc string of a callable is a reliable source to infer its signature. Even if it is, the work needed to do that correctly may not justify the benefit
2.  that said, `jedi` does try to infer function arguments for sphinx, epydoc and basic numpydoc docstrings, but it doesn't expose any public API for that.
3. PyCharm seems to rely solely on signatures to do static check/code assistant on functions

Based on the above,  I am not sure it is so bad that we just document the limitation and workaround when we can't get a signature via the standard `inspect` module.

Reply:
We could easily make this "just work" by using the return type.  Relying on documented workarounds is a poor substitute to the product functioning as expected.

Reply:
Do you mean we do a very crude parsing (regex search) of __doc__ and look for '->  type' to get the 'return type'?

Reply:
The improvement requested makes sense but is not trivial to do. We want to get the bug fix in first and #6382 is filed as an enhancement ticket for more broad inference support by the UDF parser.